x-pack/plugins/observability_solution/investigate_app/public/hooks/use_delete_investigation.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+
+import { IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
+import { useMutation } from '@tanstack/react-query';
+import { useKibana } from './use_kibana';
+
+type ServerError = IHttpFetchError<ResponseErrorBody>;
+
+export function useDeleteInvestigation({
+  onDeleteSuccess,
+  onDeleteFailure,
+}: {
+  onDeleteSuccess: () => void;
+  onDeleteFailure: () => void;
```

Comment:
We can leverage React Query query invalidation instead of passing these callbacks. That should simplify a lot of things

```
const queryClient = useQueryClient();

[...]

onSuccess: (response, {}) => {
  toast();
  queryClient.invalidateQueries({ queryKey: investigationKeys.list(), exact: false });
})
```